src/build_workflow/build_incremental.py

```diff
@@ -62,3 +68,56 @@ def rebuild_plugins(self, changed_plugins: List, input_manifest: InputManifest)
 
         logging.info(f"Rebuilding list is {rebuild_list}")
         return rebuild_list
+
+    def build_incremental(self, args: BuildArgs, input_manifest: InputManifest, components: List) -> None:
+        build_manifest_path = os.path.join(self.distribution, "builds", input_manifest.build.filename, "manifest.yml")
+        if not os.path.exists(build_manifest_path):
+            logging.error("Previous build manifest is not exists. Throw error.")
+
+        logging.info(f"Build {components} incrementally.")
+
+        build_manifest_data = BuildManifest.from_path(build_manifest_path).__to_dict__()
+
+        output_dir = BuildOutputDir(input_manifest.build.filename, args.distribution).dir
+        failed_plugins = []
+
+        with TemporaryDirectory(keep=args.keep, chdir=True) as work_dir:
+            logging.info(f"Building in {work_dir.name}")
+            target = BuildTarget(
+                name=input_manifest.build.name,
+                version=input_manifest.build.version,
+                qualifier=input_manifest.build.qualifier,
+                patches=input_manifest.build.patches,
+                snapshot=args.snapshot if args.snapshot is not None else input_manifest.build.snapshot,
+                output_dir=output_dir,
+                distribution=args.distribution,
+                platform=args.platform or input_manifest.build.platform,
+                architecture=args.architecture or input_manifest.build.architecture,
+            )
+
+            build_recorder_incremental = BuildRecorder(target, build_manifest_data)
+
+            logging.info(f"Building {input_manifest.build.name} ({target.architecture}) into {target.output_dir}")
+
+            for component in input_manifest.components.select(focus=components, platform=target.platform):
+                logging.info(f"Rebuilding {component.name}")
+
+                builder = Builders.builder_from(component, target)
+                try:
+                    builder.checkout(work_dir.name)
+                    builder.build(build_recorder_incremental)
+                    builder.export_artifacts(build_recorder_incremental)
+                    logging.info(f"Successfully built {component.name}")
+                except:
+                    logging.error(f"Error incremental building {component.name}.")
+                    if args.continue_on_error and component.name not in ['OpenSearch', 'job-scheduler', 'common-utils', 'OpenSearch-Dashboards']:
+                        failed_plugins.append(component.name)
+                        continue
+                    else:
+                        raise
+
+            build_recorder_incremental.write_manifest()
+
+        if len(failed_plugins) > 0:
+            logging.error(f"Failed plugins are {failed_plugins}")
+        logging.info("Done.")
```

Comment:
Confused as this entire block is duplicate of https://github.com/opensearch-project/opensearch-build/blob/main/src/run_build.py ? 
Can you explain why are you trying to achieve? 

Also, the build recorder object will always be either build or incremental_build. Accordingly you can initialize the object.

Adding @prudhvigodithi @dblock to take a look too.

Reply:
Just had an offline discussion with @gaiksaya. I will try to squash the build logic here and those from `run_build.py` together so that it could save us from a lot of chaos.